### ISSUE TITLE
Wrong `package.json` `workspace:` prefix behavior

### ISSUE THREAD

**Issue by iam-medvedev** [oven-sh/bun#2938]
### What version of Bun is running?

0.6.2

### What platform is your computer?

Darwin 22.3.0 arm64 arm

### What steps can reproduce the bug?

Bun cannot install linked packages with the yarn [workspace prefix](https://yarnpkg.com/features/workspaces#workspace-ranges-workspace).

`package.json`:
```json
{
  "name": "@scope/package",
  "workspaces": ["demo"]
}
```

`demo/package.json`:
```json
{
  "name": "demo",
  "private": true,
  "dependencies": {
    "@scope/package": "workspace:^"
  }
}
```

`bun install` gives the following error:
```sh
➜ bun install
bun install v0.6.2 (8d90d795)


error: package "@scope/package" is not linked

To install a linked package:
   bun link my-pkg-name-from-package-json

Tip: the package name is from package.json, which can differ from the folder name.
```

`bun link @scope/package` also gives the same error.

**Comment by Electroid** [comment]
Using Bun v1.0.7, it now shows a different error.

```
❯ bun install
bun install v1.0.7 (6a2768f2)

error: workspace dependency "@scope/package" not found

Searched in "./demo/^"

Workspace documentation: https://bun.sh/docs/install/workspaces
```

**Comment by joshmeads** [comment]
This should be able to be closed now right? v1.1.9 fixed this AFAIK

https://bun.sh/blog/bun-v1.1.9

**Comment by Jarred-Sumner** [comment]
cc @dylan-conway I'm pretty sure this has been fixed
